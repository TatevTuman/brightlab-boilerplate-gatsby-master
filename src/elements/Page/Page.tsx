import React, { memo } from 'react'
import { RouteComponentProps } from '@reach/router'
import { Container } from '@elements'
import { Children } from '@types'

export interface PageProps extends RouteComponentProps {
  children: Children
}

const Page: React.FC<PageProps> = props => {
  const { children } = props

  return (
    <div className={'page'} role="main">
      <Container>{children}</Container>
    </div>
  )
}

Page.defaultProps = {
  path: ''
}

export default memo(Page)
