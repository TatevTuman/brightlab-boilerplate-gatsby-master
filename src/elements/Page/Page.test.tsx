import React from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import Page from './Page'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Page', () => {
  const props = { children: 'children' }

  it('renders correctly', async () => {
    const { container, getByText } = render(<Page {...props} />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
    expect(getByText(props.children)).toBeInTheDocument()
  })
})
