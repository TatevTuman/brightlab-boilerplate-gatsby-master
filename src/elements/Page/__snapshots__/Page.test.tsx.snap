// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div>
  <div
    class="page"
    role="main"
  >
    <div
      class="container"
    >
      children
    </div>
  </div>
</div>
`;
